app/routes/settings/users/index.js

```diff
@@ -54,12 +128,32 @@ export default class SettingsUsersIndexRoute extends Route {
     return this.store.query('user', options);
   }
 
-  setupController(controller) {
+  async setupController(controller) {
     super.setupController(...arguments);
 
     if (controller.page !== this.lastParams.committed.page) {
       controller.page = this.lastParams.committed.page;
     }
     controller.searchTextBuffer = this.lastParams.committed.filter;
+
+    const organizationIds = this.lastParams.committed.organizations;
+    controller.selectedOrganizations = await Promise.all(organizationIds.map((id) => this.store.findRecord('user-organization', id)));
+
+    controller.dateFromBuffer = parseDate(this.lastParams.committed.dateFrom);
+    controller.dateToBuffer = parseDate(this.lastParams.committed.dateTo);
+
+    const roleIds = this.lastParams.committed.roles;
+    controller.selectedRoles = await Promise.all(roleIds.map((id) => this.store.findRecord('role', id)));
+  }
+
+  @action
+  loading(transition) {
+    // eslint-disable-next-line ember/no-controller-access-in-routes
+    const controller = this.controllerFor(this.routeName);
+    controller.isLoadingModel = true;
+    transition.promise.finally(() => {
+      controller.isLoadingModel = false;
+    });
+    return true;
```

Comment:
Nice catch that this was missing :+1: 
Should we use the same loading construct for the organizations table?